hasher-matcher-actioner/hmalib/fetching/bank_store.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from functools import reduce
+import typing as t
+
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.exchanges.signal_exchange_api import SignalExchangeAPI
+from threatexchange.exchanges.fetch_state import (
+    FetchCheckpointBase,
+    FetchDelta,
+    SignalOpinion,
+    TUpdateRecordKey,
+    TUpdateRecordValue,
+    FetchedSignalMetadata,
+)
+
+from hmalib.common.configs.tx_collab_config import EditableCollaborationConfig
+from hmalib.common.configs.tx_apis import ToggleableSignalExchangeAPIConfig
+from hmalib.common.models.bank import BanksTable
+from hmalib.common.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class NoImportAsBankDefined(Exception):
+    def __init__(self, collab_config: EditableCollaborationConfig):
+        self.collab_config = collab_config
+        super().__init__(
+            f"CollabConfig {collab_config.name} is not connected to an importable bank. Create a bank, and save again using tx_collab_config apis."
```

Comment:
fair. can convert to an assertion. I've never found a good theory of when to use exceptions and when not to for python. Any pointers?

Reply:
It comes down to the design of your modules. In general:
1. Use exceptions for exceptional events. If failure is likely and should always be handled (i.e. file.exists()), maybe return whether the operation was successful or not. 
2. Avoid `except Exception` except for outmost error handlers
3. Never `except BaseException` / `except:` unless you REALLY know what you are doing, since you will also catch things like interrupts, os.exit() and out-of-memory events, and your program is almost certainly boned or expected to close
4. Prefer existing exceptions (ValueError, KeyError) if they exactly map to what you are doing
5. If you do decide to use exceptions, create your own base class and catch that class at error boundaries.

The reason to assert here would be a combination of guidelines 1/4. We don't expect this state to ever be present in the application on purpose, so assert seems to map well to the expectations.